experimental/devices/as7262/as7262.go

```diff
@@ -125,64 +130,67 @@ func (d *Dev) Sense(ledDrive physic.ElectricCurrent, senseTime time.Duration) (S
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
-	done := make(chan struct{}, 1)
-	d.canceledMu.Lock()
-	d.done = done
-	d.canceled = false
-	d.canceledMu.Unlock()
-
 	it, integration := calcSenseTime(senseTime)
-	if err := d.writeVirtualRegister(intergrationReg, it); err != nil {
+	ctx, cancel := context.WithCancel(context.Background())
+	d.cancelMu.Lock()
+	d.cancel = cancel
+	d.done = make(chan struct{})
+	d.cancelMu.Unlock()
+	defer d.cancel()
+
```

Comment:
I don't understand why you wouldn't `defer func() { close(d.done) }()`, then Halt() wouldn't need a timeout.

Reply:
Good catch. I want to wait on the close, so Timeout was there for the case if Halt() was called before Sense() when the channel was still not closed after New(). Now that done is closed in New() that is not a problem.